.gitlab/pipeline.yml

```diff
@@ -9,6 +9,7 @@ include:
   - local: '.gitlab/includes/clang14_cuda11_pipeline.yml'
   - local: '.gitlab/includes/performance_gcc13_pipeline.yml'
   - local: '.gitlab/includes/gcc12_hip5_pipeline.yml'
+  - local: '.gitlab/includes/nvhpc23_9_pipeline.yml'
```

Comment:
While I expect NVHPC to cause "interesting" compilation failures, I think we should not trigger it automatically on pushes since, at least on the Jenkins configurations, it's by far the slowest compiler as well.

It's fine to leave this here until this is ready for merging, but I'm commenting just as a reminder that we move it down before merging.